### PR TITLE
Fix mzroll samples used

### PR DESCRIPTION
**Pull request by sahil21**

### CODE REVIEW

File: src/gui/mzroll/tabledockwidget.h [hunk 1]
```diff
@@ -180,120 +191,119 @@ public Q_SLOTS:
...
     */
-      void readSamplesXML(QXmlStreamReader &xml,PeakGroup* group);
-    /**@brief-mark varible <mzrollv_0_1_5> true or false
-     *@details  this method marks varible <mzrollv_0_1_5> true if loaded mzroll
-     * file is of v0.1.5 or older otherwise false based on one attribute
-     * <SamplesUsed> which is introduced here.
-    */
+    void readSamplesXML(QXmlStreamReader &xml, PeakGroup *group);
+
     void markv_0_1_5mzroll(QString fileName);
```
[the remark this answers is not in the document]
Comment (sahil21) [reply src/gui/mzroll/tabledockwidget.h]: @shubhra-agrawal Thanks for pointing out. I'll update

**Comment by shubhra-agrawal** [comment]
I've tested mzrolls from v0.1.5, v0.2.4 and this branch and everything seems fine. I have not been able to test untargeted csv export because it crashes and the fix is in another branch. 

Just one small point, mzroll from v0.2.4 gave me 'inf' in ppmDiff column of csv export. Not very important imo but wanted you to know.

File: src/gui/mzroll/mzfileio.cpp [hunk 2]
```diff
@@ -79,7 +80,13 @@ mzSample* mzFileIO::loadSample(QString filename){
             QString sampleNumberInfo = " | Sample Number=" + QString::number(sample->sampleNumber);
             sampleName = sampleName + sampleNumberInfo;
         }
+
         sample->sampleName = string( sampleName.toLatin1().data() );
+        
+        mtxSampleId.lock();
```
Comment (rish9511) [inline src/gui/mzroll/mzfileio.cpp]: mutex is of no use when uploading a gcms dataset ?
@sahil21  @shubhra-agrawal  
did you test this with gcms data set ?( i am on mac )

File: src/gui/mzroll/tabledockwidget.cpp [hunk 3]
```diff
@@ -1904,77 +1904,25 @@ void TableDockWidget::loadPeakTable() {
...
-        if (xml.isStartElement()) {   
-            if (xml.name() == "SamplesUsed"){
-                /**mark false if <SamplesUsed> which is only in new version*/
-                mzrollv_0_1_5=false;
-                break;
+    if (xml.name() == "SamplesUsed") {
+        xml.readNextStartElement();
+        while (xml.name() == "sample") {
+            unsigned int id = xml.attributes().value("id").toString().toInt();
+            for(int i=0;i < samples.size();++i){
```
Comment (rish9511) [inline src/gui/mzroll/tabledockwidget.cpp]: try to use range based for loops

**Pull request merged**